docs/io/unified.rst

```diff
@@ -336,6 +347,71 @@ if both a width and precision are present they are both set in the TDISPn
 format.  A Python ``f`` or ``F`` map to TDISP F format.  The Python ``g`` or
 ``G`` map to TDISP G format.  The Python ``e`` and ``E`` map to TDISP E format.
 
+Masked columns
+^^^^^^^^^^^^^^
+
+Tables that contain `~astropy.table.MaskedColumn` columns can be written to
+FITS.  By default this will replace the masked data elements with certain
+sentinel values according to the FITS standard:
+
+- ``NaN`` for float columns
+- Value of ``TNULLn`` for integer columns
+- Null string for string columns (not currently implemented)
+
+When the file is read back those elements are marked as masked in the
+returned table.
+
+The FITS standard has a few limitations:
+
+- Not all data types are supported (e.g. logical / boolean)
+- Integer columns require picking one value as the NULL indicator.  If
+  all possible values are represented in valid data (e.g. an unsigned
+  int columns with all 256 possible values in valid data) then there
+  is no way to represent missing data.
+- The masked data values are permanently lost, precluding the possibility
+  of later unmasking the values.
+
+Astropy provides a work-around for this limitation that users can choose to
+use.  The key part is setting a special attribute ``serialize_method``
+on the columns which should be written using the work-around::
+
+  >>> from astropy.table.table_helpers import simple_table
+  >>> t = simple_table(masked=True)
+  >>> t['d'] = [False, False, True]
+  >>> t['d'].mask = [True, False, False]
+  >>> t
+  <Table masked=True length=3>
+    a      b     c     d
+  int64 float64 str1  bool
+  ----- ------- ---- -----
+     --     1.0    c    --
+      2     2.0   -- False
+      3      --    e  True
+
+  >>> for col in t.itercols():
+  ...    col.info.serialize_method['fits'] = 'data_mask'
```

Comment:
Can we add a keyword argument to the Table write function for FITS files (or maybe even at a higher level so it works for HDF5 and ECSV) so that one can do:

```python
t.write('myfile.fits', serialize_method='data_mask')
```

I think this would be cleaner for users than having to loop as above.

Reply:
Agreed the interface is a little clunky and maybe we can come up with something better.  The problem is that we are dealing with different formats that have different requirements and format-specific rules and defaults.  tl;dr - it's a little complicated, and basically instead of thinking to invent something elegant I just shoe-horned onto the existing interface.

`serialize_method` came into being as a way to specify that, by default, a `Time` object should be serialized to most ASCII table formats (including ECSV) as the formatted value, but for FITS and HDF5 it should be a jd1/jd2 tuple.  The code doing the conversion would not necessarily know the context (FITS or ECSV), so we put in a context method to provide that.  So:

1. `serialize_method` defines default reasonable behaviors for column types that depend on the format (FITS, HDF5, etc).  Previously there was only `Time`, but now also `MaskedColumn`.
2. `serialize_method` exists as an Info attribute that can be customized.

Within a single table you can have `Time` and `MaskedColumn` objects, so any high-level `serialize_method` needs to allow a dict keyed by column name.  Not a problem, though if you have a masked table with Time columns you will need to fill in the dict appropriately for every column anyway.

One way forward is to open an issue to define a better high-level interface.  That is effectively independent of this PR, because no matter what you will need to retain the `Info.serialize_method` attribute for the internal code.  This provides a very convenient way to get this information down from the user interface into the low-level routines that need it.  (And of course it is already in the public API for Time).  If there is a better UI that could be what people see in the user docs.

Reply:
This makes sense. Just to clarify, we could implement this keyword argument *just* for the FITS writer though.